packages/svelte/src/internal/client/runtime.js

```diff
@@ -191,9 +191,36 @@ function create_source_signal(flags, value) {
  * @param {import('./types.js').SignalFlags} flags
  * @param {V} value
  * @param {import('./types.js').Block | null} block
- * @returns {import('./types.js').ComputationSignal<V>}
+ * @returns {import('./types.js').ComputationSignal<V> | import('./types.js').ComputationSignal<V> & import('./types.js').SourceSignalDebug}
  */
 function create_computation_signal(flags, value, block) {
+	if (DEV) {
+		return {
+			// block
+			b: block,
+			// consumers
+			c: null,
+			// destroy
+			d: null,
+			// equals
+			e: null,
+			// flags
+			f: flags,
+			// init
+			i: null,
+			// references
+			r: null,
+			// value
+			v: value,
+			// context: We can remove this if we get rid of beforeUpdate/afterUpdate
+			x: null,
+			// destroy
+			y: null,
+			// this is for DEV only
+			inspect: new Set()
+		};
+	}
+
```

Comment:
My initial thought was to do this...

```js
if (DEV) {
  signal.inspect = ...
}
```

...but the reason we don't do that is that you're changing the object shape when you do that, which IIUC (though @trueadm is the authority on such things) makes subsequent reads slower.

The spread approach should avoid that problem, though it _does_ mean that you have to create the signal twice. It's not the worst thing in the world since it's only in dev, but ideally dev would resemble prod as closely as is practical. I'll defer to @trueadm on this question.

Reply:
Yeah, I think this acceptable.